src/spikeinterface/postprocessing/tests/common_extension_tests.py

```diff
@@ -61,65 +61,98 @@ def get_sorting_analyzer(recording, sorting, format="memory", sparsity=None, nam
+        """
+        This method sets up the class once at the start of testing. It is
+        in scope for the lifetime of te class and is reused across all
+        tests that inherit from this base class to save processing time and
+        force a small radius.
+
+        When setting attributes on `self` in `scope="class"` a new
+        class instance is used for each. In this case, we have to set
+        from the base object `__class__` to ensure the attributes
+        are available to all subclass instances.
+        """
+        self.__class__.recording, self.__class__.sorting = get_dataset()
+
+        self.__class__.sparsity = estimate_sparsity(
+            self.__class__.recording, self.__class__.sorting, method="radius", radius_um=20
+        )
 
-    def _prepare_sorting_analyzer(self, format, sparse):
-        # prepare a SortingAnalyzer object with depencies already computed
+    def _prepare_sorting_analyzer(self, format, sparse, extension_class):
+        """
+        Prepare a SortingAnalyzer object with dependencies already computed
+        according to format (e.g. "memory", "binary_folder", "zarr")
+        and sparsity (e.g. True, False).
+        """
         sparsity_ = self.sparsity if sparse else None
+
         sorting_analyzer = get_sorting_analyzer(
-            self.recording, self.sorting, format=format, sparsity=sparsity_, name=self.extension_class.extension_name
+            self.recording, self.sorting, format=format, sparsity=sparsity_, name=extension_class.extension_name
         )
         sorting_analyzer.compute("random_spikes", max_spikes_per_unit=50, seed=2205)
-        for dependency_name in self.extension_class.depend_on:
+
+        for dependency_name in extension_class.depend_on:
             if "|" in dependency_name:
                 dependency_name = dependency_name.split("|")[0]
             sorting_analyzer.compute(dependency_name)
+
         return sorting_analyzer
 
-    def _check_one(self, sorting_analyzer):
-        if self.extension_class.need_job_kwargs:
+    def _check_one(self, sorting_analyzer, extension_class, params):
+        """
+        Take a prepared sorting analyzer object, compute the extension of interest
+        with the passed parameters, and check the output is not empty, the extension
+        exists and `select_units()` method works.
+        """
+        if extension_class.need_job_kwargs:
             job_kwargs = dict(n_jobs=2, chunk_duration="1s", progress_bar=True)
         else:
             job_kwargs = dict()
 
-        for params in self.extension_function_params_list:
-            print("  params", params)
-            ext = sorting_analyzer.compute(self.extension_name, **params, **job_kwargs)
-            assert len(ext.data) > 0
-            main_data = ext.get_data()
+        ext = sorting_analyzer.compute(extension_class.extension_name, **params, **job_kwargs)
+        assert len(ext.data) > 0
+        main_data = ext.get_data()
```

Comment:
Could add a test to check that main_data is non-empty? At the moment we do nothing with it.
```suggestion
        assert len(main_data) > 0
```